enhancements/installer/aws-cross-account-dns-zone.md

```diff
@@ -0,0 +1,340 @@
+appropriate resource and a trust policy to allow access from a second account.
+
+An example IAM Permissions Policy, which grants permissions to the private hosted zone:
+
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+                "route53:GetHostedZone",
+                "route53:ChangeResourceRecordSets",
+                "route53:ChangeTagsForResource"
+            ],
+            "Resource": [
+                "arn:aws:route53:::hostedzone/Z00147933I3NWOQ6M4699"
+            ]
+        }
+    ]
+}
+```
+
+An example IAM Trust Policy:
+
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Principal": {
+                "AWS": "arn:aws:iam::[ID for Account B]:user/<user name for cluster creator>"
+            },
+            "Action": "sts:AssumeRole",
+            "Condition": {}
+        }
+    ]
+} 
+
+```
+
+### Risks and Mitigations
+
+Add-on operators needing permissions will not work unless updated to use the role ARN.
+
+### Drawbacks
+
+Up to this point, DNS config has been platform agnostic. There are no obvious drawbacks to introducing
+platform dependencies in this config, but it should be carefully considered.
+
+## Design Details
+
+### Open Questions [optional]
+
+1. The suggested field name for the API is clunky, but has the advantage of being least ambiguous.
+Some other possible candidates would be:
+- sharedVPCRole
+- privateDNSRole
+- privateZoneRole
+- privateDNSZoneRole
+- dnsRole
+- dnsZoneRole
+- privateHostedZoneRole
+2. In the install config, should we consider deprecating `aws.platform.hostedZone` in order to
+create a struct that can nest both the `id` and `role` fields? Using the same example as above,
+the install config would look like:
+
+```yaml
+platform:
+  aws:
+    region: us-east-2
+    subnets:
+    - subnet-0ef71407239a73558
+    - subnet-00e576c91f8578661
+    - subnet-041af6e1bbf29c2ef
+    - subnet-037ab7dc0cd396721
+    - subnet-0adda52a1007fc74f
+    - subnet-09bd88678851acab6
+    hostedZone: #deprecated
+    privateHostedZone:
+      id: Z0743278N5ELJPLYA3H6
+      role: arn:aws:iam::<account-a>:role/<role-name>
+```
+
+3. Should cluster admins be able to edit the API object day 2? In general, it does not seem like
+this should be a primary goal, but we do have one instance where an organization has a support
+exception to enable this functionality through manual intervention. It could be possible for UPI
+clusters as well. In which case there may be limited demand for upgrading clusters to be able to 
+update the API object in order to have the ingress operator manage records.
+
+### Test Plan
+
+An e2e test will be setup which utilizes both AWS CI accounts.
+
+### Graduation Criteria
+
+This functionality is targeted for 4.14 GA and for backporting to previous releases. 
```

Comment:
Have you considered a feature gate? 

Do we often backport features? Especially when they include API changes as described here?

Reply:
Added details on feature gate.

> Do we often backport features? Especially when they include API changes as described here?

No, but this is being prioritized as an exception.